### PR TITLE
add es7210 component (BSP-215)

### PR DESCRIPTION
**Pull request by L-KAYA**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [x] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [x] New files were added to CI build job
- [x] _Optional:_ Component contains unit tests
- [x] CI passing

# Change description
Add a new  component `es7210`: a 4-channel audio ADC

### CODE REVIEW

File: components/es7210/idf_component.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version: 1.0.0-alpha
```
Comment (tore-espressif) [inline components/es7210/idf_component.yml:1]: Could we start with 1.0.0? Or do you expect any changes in foreseeable time?

File: test_app/CMakeLists.txt [hunk 2]
```diff
@@ -13,12 +13,13 @@ if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_GREATER_EQUAL "4.4")
 endif()
 if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_LESS "4.4")
     set(EXCLUDE_COMPONENTS "es8311")
+    set(EXCLUDE_COMPONENTS "es7210")
```
Comment (tore-espressif) [inline test_app/CMakeLists.txt:16]: are are resetting the EXCLUDE_COMPONENTS variable here.

Could you add it to the same line as es8311?

**Review by tore-espressif — APPROVED** [review]
@L-KAYA Thank you very much for this complete PR!

Could you please check your formatting? We have pre-commit in place, that will automatically format your code.

Otherwise LGTM!

File: components/es7210/idf_component.yml (hunk 1 above)
Comment (L-KAYA) [reply components/es7210/idf_component.yml:1]: Sure

File: test_app/CMakeLists.txt (hunk 2 above)
Comment (L-KAYA) [reply test_app/CMakeLists.txt]: Done. Sorry for the mistake.

**Comment by tore-espressif** [comment]
@L-KAYA seems like `#include <inttypes.h>` is missing in es7210.c?

**Comment by L-KAYA** [comment]
Yes.. Forgot it is not included in v4.4.

**Pull request merged**